### PR TITLE
Cleanup: WorkTree/Index should be used rather than Unstaged/Staged

### LINKED ISSUE #5230: Cleanup: WorkTree/Index should be used rather than Unstaged/Staged
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
Code has inconsistent use of WorkTree and Index "artificial" commits.
Unstaged/Staged are are used too.
The most common combination is Unstaged/Index

The Git documentation has been cleaned up to consistently use WorkTree/Index, the is what should be used in the code too

Note: Stage/Unstage can still be used as the action to move files to/from WorkTree/Index

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
n/a

**What is the expected behavior?**
Consistency with Git documentation

**Environment you encounter the issue:**

- GitExtensions version: all

**Did this work in previous version of GitExtensions (which)?**
no

### PR DESCRIPTION
**Pull request by gerhardol**
Fixes #5230 

Changes proposed in this pull request:
- Internal renaming in the code:
 WorkTree/Index should be used rather than Unstaged/Staged
 
Screenshots before and after (if PR changes UI):
- n/a

What did I do to test the code and ensure quality:
- no changes

Has been tested on (remove any that don't apply):

### CODE REVIEW

File: GitUI/CommandsDialogs/RevisionDiffControl.cs [hunk 1]
```diff
@@ -273,8 +273,8 @@ private ContextMenuSelectionInfo GetSelectionInfo()
                 isAnyCombinedDiff: isAnyCombinedDiff,
                 isSingleGitItemSelected: isExactlyOneItemSelected,
                 isAnyItemSelected: isAnyItemSelected,
-                isAnyItemStaged: isAnyStaged,
-                isAnyItemUnstaged: isAnyUnstaged,
+                isAnyItemIndex: isAnyIndex,
+                isAnyItemWorkTree: isAnyWorkTree,
```
Comment (drewnoakes) [inline GitUI/CommandsDialogs/RevisionDiffControl.cs]: `isAnyItemInIndex` / `isAnyItemInWorkTree`?

File: GitUI/UserControls/RepoStateVisualiser.cs [hunk 2]
```diff
@@ -29,20 +29,20 @@ internal sealed class RepoStateVisualiser : IRepoStateVisualiser
...
                 if (status.Staged == StagedStatus.Index)
                 {
-                    stagedCount++;
+                    indexCount++;
                 }
 
                 if (status.Staged == StagedStatus.WorkTree && status.IsSubmodule)
                 {
-                    unstagedSubmodulesCount++;
+                    workTreeSubmodulesCount++;
```
Comment (drewnoakes) [inline GitUI/UserControls/RepoStateVisualiser.cs]: I find this case a bit harder to understand. `unstagedSubmodule` makes it clear what's the adjective and what's the noun. `workTreeSubmodule` sounds like one noun. Is it the work tree of the submodule? Maybe `workTreeSubmoduleChangeCount`? Same for `indexCount`. There is only one index. `indexChangeCount`?

File: GitUI/CommandsDialogs/RevisionDiffControl.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/CommandsDialogs/RevisionDiffControl.cs]: better

File: GitUI/UserControls/RepoStateVisualiser.cs (hunk 2 above)
Comment (gerhardol) [reply GitUI/UserControls/RepoStateVisualiser.cs]: workTreeChangeSubmoduleCount and indexChangeCount

**Comment by gerhardol** [comment]
It would have been better if Git had aligned on Unstaged/Stage, better following the act of Staging files. 
But it is easier to change GE than Git
Developers need to adapt too

Not very important, but this has bugged me for some time and #5087 is finally merged!

**Comment by RussKie** [comment]
@gerhardol you will have to toss a coin with @pmiossec who's going in first - #5137 uses the old naming

**Comment by pmiossec** [comment]
@RussKie I don't see where you see a problem with #5137. But my PR is simple so I could update it if needed...

**Comment by RussKie** [comment]
In this PR Gerhard renamed all "staged/unstaged" strings to "index/worktree".
If you're ok to update your PR to align with the changes in this PR, then I will merge this one first, then get yours in.

**Comment by pmiossec** [comment]
I'm ok. But the 2 only ref I used seems to not have been renamed.
If I am wrong, I will do the changes...

**Pull request merged**